pkg/frontend/admin_hive_clusterdeployment_get_test.go

```diff
@@ -0,0 +1,116 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/env"
+	"github.com/Azure/ARO-RP/pkg/hive"
+	"github.com/Azure/ARO-RP/pkg/metrics/noop"
+	mock_env "github.com/Azure/ARO-RP/pkg/util/mocks/env"
+	mock_hive "github.com/Azure/ARO-RP/pkg/util/mocks/hive"
+)
+
+func Test_getAdminHiveClusterDeployment(t *testing.T) {
+	fakeUUID := "00000000-0000-0000-0000-000000000000"
+	ctx := context.Background()
+	clusterDeployment := hivev1.ClusterDeployment{
+		Spec: hivev1.ClusterDeploymentSpec{
+			ClusterName: "abc123",
+		},
+	}
+	type test struct {
+		name       string
+		resourceID string
+		properties api.OpenShiftClusterProperties
+		mocks      func(t *test, clusterManager *mock_hive.MockClusterManager)
+
+		wantStatusCode int
+		wantResponse   []byte
+		wantError      string
+	}
+
+	for _, tt := range []*test{
+		{
+			name:       "cluster has hive profile with namespace",
+			resourceID: fmt.Sprintf("/subscriptions/%s/resourcegroups/resourceGroup/providers/Microsoft.RedHatOpenShift/openShiftClusters/hive", fakeUUID),
+			properties: api.OpenShiftClusterProperties{HiveProfile: api.HiveProfile{Namespace: fmt.Sprintf("aro-%s", fakeUUID)}},
+			mocks: func(t *test, clusterManager *mock_hive.MockClusterManager) {
+				clusterManager.EXPECT().GetClusterDeployment(gomock.Any(), gomock.Any()).Return(&clusterDeployment, nil).Times(1)
+			},
```

Comment:
setting up the mocks should be done in the test, not here. it was a bad pattern and I'd rather not add new tests with that pattern

Reply:
I've removed the mocks from the test struct.